app/src/main/java/com/example/internetshop/presentation/activity/MainActivity.kt

```diff
@@ -41,48 +42,57 @@ class MainActivity : AppCompatActivity(), ContainerHolder {
         }
     }
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (this.applicationContext as InternetshopApplication).appComponent.inject(this)
+        binding = DataBindingUtil.setContentView(this,R.layout.activity_main)
+        setContentView(binding?.root)
+        binding?.let {
+            it.viewModel = titleViewModel
+            it.lifecycleOwner = this
+        }
+
+        setupAppBar()
+        setupBackArrowVisibility()
+        setScrollingView()
+        updateLayoutParams()
+        startFragment()
+    }
+
     override fun onStop() {
         super.onStop()
         binding?.appBar?.removeOnOffsetChangedListener(offSetListener)
     }
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        WindowCompat.setDecorFitsSystemWindows(window, false)
-        (this.applicationContext as InternetshopApplication).appComponent.inject(this)
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        val appBarLayoutParams = binding?.appBar?.layoutParams as CoordinatorLayout.LayoutParams
-        if (appBarLayoutParams.behavior == null) {
-            appBarLayoutParams.behavior = AppBarLayout.Behavior()
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.home -> {
+                supportFragmentManager.popBackStack()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
         }
-        setContentView(binding?.root)
-        setSupportActionBar(binding?.toolbar)
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
-        supportActionBar?.setDisplayShowHomeEnabled(true)
-        supportActionBar?.setHomeAsUpIndicator(
-            (ResourcesCompat.getDrawable(resources, R.drawable.ic_back_arrow, null))
-        )
+    }
+
+    private fun startFragment() {
         binding?.let {
-            val fragment = AuthenticationFragment()
             supportFragmentManager.beginTransaction()
-                .replace(it.fragmentContainer.id, fragment)
+                .replace(it.fragmentContainer.id, AuthenticationFragment())
                 .commit()
         }
-        ViewCompat.setOnApplyWindowInsetsListener(binding?.appBar!!) { view, windowInsets ->
-            val insets = windowInsets.getInsets(WindowInsetsCompat.Type.systemGestures())
-            val params = view.layoutParams as CoordinatorLayout.LayoutParams
-            params.topMargin = insets.top
-            view.layoutParams = params
-            offSetListener?.updateInsets(insets)
-            WindowInsetsCompat.CONSUMED
-        }
-        titleViewModel.titleLiveData.observe(this, {
-            binding?.collapsingLayout?.title = it
-        })
+    }
+
+    private fun setupBackArrowVisibility() {
```

Comment:
Would be better to name it like `subscribeToBackArrowVisibility` or `listenBackArrowVisibility` just to provide more info about the method. Coz you don't setup visibility once. You listen for changes and react for them

Reply:
fixed